### ISSUE TITLE
ContainSingle() should list the actual items in the enumerable on failure

### ISSUE THREAD

**Issue by jon-armen** [fluentassertions/fluentassertions#436]
First, thank you for fixing the issue with ContainSingle not honoring the reason.
It would be beneficial if the actual values in the subject were returned also returned.  Short of testing, I believe this code should do the trick

```
public AndWhichConstraint<TAssertions, T> ContainSingle(string because = "", params object[] becauseArgs) 
{ 
    if (ReferenceEquals(Subject, null)) 
    { 
        Execute.Assertion 
            .BecauseOf(because, becauseArgs) 
            .FailWith("Expected {context:collection} to contain a single item{reason}, but found <null>."); 
    } 

    switch(Subject.Count())
    {
        case 0: // Fail, Collection is empty
            Execute.Assertion.BecauseOf(because, becauseArgs).FailWith("Expected {context:collection} to contain a single item{reason}, but the collection is empty.");
            break;
        case 1: // Success Condition
            break;
        default: // Fail, Collection contains more than a single item
            Execute.Assertion.BecauseOf(because, becauseArgs).FailWith("Expected {context:collection} to contain a single item{reason}, but found {0}.", Subject);
            break;
    }

    return new AndWhichConstraint<TAssertions, T>((TAssertions)this, Subject.Single()); 
}
```

**Comment by dennisdoomen** [comment]
Nice. Would you be willing to create a PR for this?

**Comment by jon-armen** [comment]
Sure

**Comment by kelgrims** [comment]
Hi, is this still up for grabs?

**Comment by dennisdoomen** [comment]
@kelgrims probably. Haven't received any PRs from @jon-armen AFAIK.

**Comment by kelgrims** [comment]
@jon-armen are you going to submit a PR or should I?

**Comment by jon-armen** [comment]
@kelgrims; @dennisdoomen  I was going to make a PR, but ended up getting pulled in several other directions at home.  Feel free to go ahead and make the PR for this. 

Thanks, Jon

**Comment by jnyrup** [comment]
I believe #497 closes this.

**Comment by matthiaslischka** [comment]
I'm confused here. The title suggests that all items of the collection should be listed in the error message. But the fix - for me (Version 5.9.0) - only displays the collection name. Not it's content.

I would assume the title was meant that `ContainSingle` lists the Collection entries just like `Contain` does.

All I get is
> Expected validationResult.Errors to contain a single item matching x.ErrorMessage.Contains("foobar"), but no such item was found.

**Comment by jnyrup** [comment]
@matthiaslischka The fix provided here was for 
```c#
ContainSingle(string because = "", params object[] becauseArgs)
```

Your case is for the other overload
```c#
ContainSingle(Expression<Func<T, bool>> predicate,string because = "", params object[] becauseArgs)
```
